### DISCUSSION TITLE
Have a request send credentials for windows authorization

### DISCUSSION THREAD

**Discussion by phoopee3** [jhuckaby/Cronicle#557] [General]
## Summary

I am running cronicle on my local computer, and want it to ping some servers I have at work. The issue is that the servers I have are running Windows and IIS. On top of that, the sites are locked down using "windows authentication". When I use other api tools, like postman, insomnia or cURL, I have to use an authentication method called NTLM to pass the credentials with the request. I cannot find a way to include those credentials in the request. If someone can tell me where the request is actually made, or what tool/script it's using, maybe I can find a way to tack on the username and password.

## Steps to reproduce the problem

Have a server running Windows, IIS, and a site set up in that that has basic authentication turned off, and windows authentication turned on.

Use another computer running cronicle to try to ping that server.

## Your Setup

I have the basic cronicle set up on my Mac.

### Operating system and version?

Mac, 12.6.1
Server, Windows server 2012 (don't judge me), IIS 8 (really please don't judge me)

### Node.js version?

18.7.0

### Cronicle software version?

0.9.16

### Are you using a multi-server setup, or just a single server?

Single server

### Are you using the filesystem as back-end storage, or S3/Couchbase?

File system

**Comment by jhuckaby** [comment]
Hmmm... well, the Cronicle [HTTP Request Plugin](https://github.com/jhuckaby/Cronicle/blob/master/docs/Plugins.md#built-in-http-request-plugin) probably cannot do what you want.  It's only designed to make simple HTTP requests. 
 What you want to use is `curl`.  I'd suggest using the [Shell Plugin](https://github.com/jhuckaby/Cronicle/blob/master/docs/Plugins.md#built-in-shell-plugin), and using the `curl` utility that is built-in to your Mac.  Example:

```sh
#!/bin/bash

curl http://enterprisesharepoint -v --ntlm --negotiate -u USER123:PASSWORD123
```

I'm going to migrate this over to a discussion, as it is not an issue with Cronicle per se.

**Reply by phoopee3** [reply]
OK thank you for the quick reply. I will see what I can do with that!
